test/groovy/MulticloudDeployTest.groovy

```diff
@@ -276,4 +289,81 @@ class MulticloudDeployTest extends BasePiperTest {
+
+        stepRule.step.multicloudDeploy([
+            script                      : nullScript,
+            enableZeroDowntimeDeployment: true,
+            parallelExecution           : true,
+            source                      : 'file.mtar'
+        ])
+
+        assertTrue(executedOnKubernetes)
+        assertFalse(executedOnNode)
+
+        assert cloudFoundryDeployRule.hasParameter('script', nullScript)
+        assert cloudFoundryDeployRule.hasParameter('deployType', 'blue-green')
+        assert cloudFoundryDeployRule.hasParameter('cloudFoundry', cloudFoundry1)
+        assert cloudFoundryDeployRule.hasParameter('mtaPath', nullScript.commonPipelineEnvironment.mtarFilePath)
+        assert cloudFoundryDeployRule.hasParameter('deployTool', 'cf_native')
+
+        assert cloudFoundryDeployRule.hasParameter('script', nullScript)
+        assert cloudFoundryDeployRule.hasParameter('deployType', 'blue-green')
+        assert cloudFoundryDeployRule.hasParameter('cloudFoundry', cloudFoundry2)
+        assert cloudFoundryDeployRule.hasParameter('mtaPath', nullScript.commonPipelineEnvironment.mtarFilePath)
+        assert cloudFoundryDeployRule.hasParameter('deployTool', 'cf_native')
+
+    }
+
+    @Test
+    void multicloudParallelOnNodeTest() {
+        stepRule.step.multicloudDeploy([
+            script                      : nullScript,
+            enableZeroDowntimeDeployment: true,
+            parallelExecution           : true,
+            source                      : 'file.mtar'
+        ])
+
+        assertTrue(executedOnNode)
+        assertFalse(executedOnKubernetes)
+
+        assert cloudFoundryDeployRule.hasParameter('script', nullScript)
+        assert cloudFoundryDeployRule.hasParameter('deployType', 'blue-green')
+        assert cloudFoundryDeployRule.hasParameter('cloudFoundry', cloudFoundry1)
+        assert cloudFoundryDeployRule.hasParameter('mtaPath', nullScript.commonPipelineEnvironment.mtarFilePath)
+        assert cloudFoundryDeployRule.hasParameter('deployTool', 'cf_native')
+
+        assert cloudFoundryDeployRule.hasParameter('script', nullScript)
+        assert cloudFoundryDeployRule.hasParameter('deployType', 'blue-green')
+        assert cloudFoundryDeployRule.hasParameter('cloudFoundry', cloudFoundry2)
+        assert cloudFoundryDeployRule.hasParameter('mtaPath', nullScript.commonPipelineEnvironment.mtarFilePath)
+        assert cloudFoundryDeployRule.hasParameter('deployTool', 'cf_native')
+    }
+
+    @Test
+    void multicloudParallelCfStandardDeployTest() {
+        stepRule.step.multicloudDeploy([
+            script                      : nullScript,
+            enableZeroDowntimeDeployment: false,
+            parallelExecution           : true,
+            source                      : 'file.mtar'
+        ])
+
+        assertFalse(executedOnNode)
+        assertFalse(executedOnKubernetes)
+
+        assert cloudFoundryDeployRule.hasParameter('script', nullScript)
```

Comment:
same here, but I am missing an assertion that checks that the deployments run in parallel

Reply:
Yes that's right I'll add it.